docs/using/syntax-optional.md

```diff
@@ -1,32 +1,202 @@
+
+- `'single quotes'`: 'single quotes'
+- `"double quotes"`: "double quotes"
+
+Adding `"replacements"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will automatically convert some common typographic texts
+
+text  | converted
+----- | ----------
+``(c)``, ``(C)`` | (c)
+``(tm)``, ``(TM)`` | (tm)
+``(r)``, ``(R)`` | (r)
+``(p)``, ``(P)`` | (p)
+``+-`` | +-
+``...`` | ...
+``?....`` | ?....
+``!....`` | !....
+``????????`` | ????????
+``!!!!!`` | !!!!!
+``,,,`` | ,,,
+``--`` | --
+``---`` | ---
+
+(syntax/linkify)=
+
+## Linkify
+
+Adding `"linkify"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will automatically identify "bare" web URLs and add hyperlinks:
+
+`www.example.com` -> www.example.com
+
+:::{important}
+This extension requires that [linkify-it-py](https://github.com/tsutsu3/linkify-it-py) is installed.
+Either directly; `pip install linkify-it-py` or *via* `pip install myst-parser[linkify]`.
+:::
+
+(syntax/substitutions)=
+
+## Substitutions
+
+Adding `"substitution"` to `myst_enable_extensions` (in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html)) will allow you to add substitutions, added in either the `conf.py` using `myst_substitutions`:
+
+```python
+myst_substitutions = {
+  "key1": "I'm a **substitution**"
+}
+```
+
+or at the top of the file, in the front-matter section (see [this section](syntax/frontmatter)):
+
+````yaml
+---
+substitutions:
+  key1: "I'm a **substitution**"
+  key2: |
+    ```{note}
+    {{ key1 }}
+    ```
+  key3: "![fun fish](img/fun-fish.png)"
+---
+````
+
+:::{important}
+Keys in the front-matter will override ones in the `conf.py`.
+:::
 
-A special syntax for admonitions can optionally be enabled by setting `myst_admonition_enable = True` in the sphinx `conf.py` [configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html).
+You can use these substitutions inline or as blocks, and you can even nest substitutions in other substitutions:
 
-The key differences are that, instead of back-ticks, colons are used, and **the content starts as regular Markdown**.
-This has the benefit of allowing the content to be rendered correctly, when you are working in any standard Markdown editor.
-For example:
+:::{tabbed} Markdown Input
 
+```md
+Inline: {{ key1 }}
+
+{{ key2 }}
+
+| col1     | col2     |
+| -------- | -------- |
+| {{key3}} | {{key3}} |
+
+```
+
+:::
+
+:::{tabbed} Rendered Output
+Inline: {{ key1 }}
+
+{{ key2 }}
+
+| col1     | col2     |
+| -------- | -------- |
+| {{key3}} | {{key3}} |
+
+:::
+
+:::{important}
+
+Substitutions will only be assessed where you would normally use Markdown, e.g. not in code blocks:
+
+````
+```
+{{ key1 }}
+```
+````
+
+```
+{{ key1 }}
+```
+
+:::
+
+(syntax/colon_fence)=
+
+## Code fences using colons
```

Comment:
I think this makes sense because having the `:::` behave similarly-but-not-quite the same was adding confusion. That said, I think we should make sure to have good "best practice" guides for people - as someone noted in another issue, we're now adding *so much* functionality or flexibility that it can be hard for people to figure out the "right" way to go about doing things.

Reply:
As I explained there, I think this is generally already done, but happy to review if someone wants to try altering the Jupyter book documentation.
In the actual documentation here, I think its pretty clear, whereby there is one file/page for (core) syntax; `docs/using/syntax.md` and one for optional/extended syntax; `docs/using/syntax-optional.md`.
There are then one "core" way an other optional ways, that pepoole can ignore if they want.

The main thing really, is if you care about what the source files look like. If not, just use core, but if you want it to be most readable outside of sphinx (e.g. in notebooks) then use things like `:::` blocks and HTML images

Reply:
I think that here it's totally fine - the myst parser docs are more "low level" and thus I think should focus on explaining all the features in full over streamlining for a particular workflow. For Jupyter Book perhaps we'll want to re-visit the "introductory" parts a bit more, since that's a more "opinionated" distribution and will probably attract more beginner-level readers that are more likely to get tripped up. I don't think this is actionable on this PR at all though 👍